### PR TITLE
Use BASE_UPSTREAM_CHECKSUM to keep track of the original base image checksum

### PR DESCRIPTION
**Pull request by aandvalenzuela**
As discussed in https://github.com/cms-sw/cms-docker/pull/245#issuecomment-1866266018

### CODE REVIEW

File: bin/docker-images-has-changed.py [hunk 1]
```diff
@@ -38,15 +38,25 @@ def find_image_deps(image, cache):
...
-  for img in get_docker_images(reponame):
+  images_list = get_docker_images(reponame)
+
+  # Generate a dictionary of base image checksum
+  base_dict = {}
+  for element in images_list:
+    name = element["BASE_DOCKER_REPOSITORY"]
+    arch = element["ARCHITECTURE"]
+    if name != "cmssw":
+        base_dict[arch] = element["BASE_UPSTREAM_CHECKSUM"]
```
Comment (smuzaffar) [inline bin/docker-images-has-changed.py:49]: I think once we have found a non-cmssw container then we can break the loop ... right?

Comment (aandvalenzuela) [reply bin/docker-images-has-changed.py:49]: Mmm... but we need to populate `base_dict` for all archs

File: bin/get_image_config.py [hunk 2]
```diff
@@ -97,6 +97,7 @@ def process_tags(setup, data, images):
     images[-1]['IMAGE_TAG']=get_key('tag', img_data)
     images[-1]['IMAGE_TAG_ALIAS']=get_key('alias', img_data)
     images[-1]['MULTI_ARCH']=get_key('multi_arch', img_data)
+    images[-1]['BASE_UPSTREAM_CHECKSUM']=from_manifest
```
Comment (smuzaffar) [inline bin/get_image_config.py:100]: `from_manifest` points to base image e.g. for cmssw/el8:grid the `from_manifest` will be `cmssw/el8:runtime` ... right?

File: bin/docker-images-has-changed.py (hunk 1 above)
Comment (smuzaffar) [reply bin/docker-images-has-changed.py:49]: ah ok, right

File: bin/get_image_config.py (hunk 2 above)
Comment (aandvalenzuela) [reply bin/get_image_config.py:100]: I am looking into it

Comment (aandvalenzuela) [reply bin/get_image_config.py:100]: Ah, yes! `from_manifest` points to the base image in each case. Then in https://github.com/aandvalenzuela/cms-docker/blob/track-base/bin/docker-images-has-changed.py#L44 we construct a dictionary only with the checksum values of the `almalinux` images and we add the corresponding `almalinux` base checksum to the property file depending on the arch

**Pull request merged**